site/src/contexts/ProxyContext.tsx

```diff
@@ -0,0 +1,185 @@
+import { useQuery } from "@tanstack/react-query"
+import { getApplicationsHost, getWorkspaceProxies } from "api/api"
+import { Region } from "api/typesGenerated"
+import { useDashboard } from "components/Dashboard/DashboardProvider"
+import { createContext, FC, PropsWithChildren, useContext, useState } from "react"
+
+interface ProxyContextValue {
+  proxy: PreferredProxy
+  isLoading: boolean
+  error?: Error | unknown
+  setProxy: (regions: Region[], selectedRegion: Region | undefined) => void
+}
+
+interface PreferredProxy {
+  // SelectedRegion is the region the user has selected.
+  // Do not use the fields 'path_app_url' or 'wildcard_hostname' from this
+  // object. Use the preferred fields.
+  selectedRegion: Region | undefined
+  // PreferredPathAppURL is the URL of the proxy or it is the empty string
+  // to indicte using relative paths. To add a path to this:
+  //  PreferredPathAppURL + "/path/to/app"
```

Comment:
If this is the case we should make sure we strip the trailing slash when we populate the context below.

Reply:
I do!:
https://github.com/coder/coder/blob/stevenmasley%2Fregions/site/src/contexts/ProxyContext.tsx#L160